src/tests/src/expressions/test_evaluate-sum-operators.cpp

```diff
@@ -0,0 +1,364 @@
+// - the following expression : sum(0 .. 1, p) is rejected.
+// - sum(t .. t+1, p) means that for each time step t, we sum p over the interval [t, t+1].
+// - sum(t .. t+1, p) current evaluation result has the same dimension as p.
+//   Please see following tests.
+// In code, time sum operator is represented by type TimeSumNode.
+// ==================================================================
+BOOST_AUTO_TEST_SUITE(tests_on_sum_over_interval_t__t_plus_1)
+
+BOOST_FIXTURE_TEST_CASE(sum_a_parameter_as_time_series_on_interval_t__t_plus_1, tests_fixture)
+{
+    // Expression : sum(t .. t+1, p), where p = {p1, p2, p3} = {1., 2., 3.}
+    Node* p = parameter("p", VariabilityType::VARYING_IN_TIME_ONLY);
+
+    Node* from = literal(0);
+    Node* to = literal(1);
+    Node* sum = timeSum(from, to, p);
+
+    auto evalResult = evalVisitor->dispatch(sum);
+
+    // Expected evaluation result : (p1 + p2, p2 + p3, p3 + p1) = (3., 5., 4.).
+    std::vector<double> expected = {3., 5., 4.};
+    std::vector<double> actual = evalResult.valuesAsVector();
+    BOOST_CHECK(actual == expected);
+}
+
+BOOST_FIXTURE_TEST_CASE(sum_a_squared_param_as_TS_on_interval_t__t_plus_1, tests_fixture)
+{
+    // Expression : sum(t .. t+1, p^2), where p = {p1, p2, p3} = {1., 2., 3.}
+    Node* p = parameter("p", VariabilityType::VARYING_IN_TIME_ONLY);
+
+    Node* from = literal(0);
+    Node* to = literal(1);
+    Node* sum_of_squares = timeSum(from, to, square(p));
+
+    auto evalResult = evalVisitor->dispatch(sum_of_squares);
+
+    // Expected evaluation result : (p1^2 + p2^2, p2^2 + p3^2, p3^2 + p1^2)
+    // = (1^2 + 2^2, 2^2 + 3^2, 3^2 + 1^2) = (5., 13., 10.)
+    std::vector<double> expected = {5., 13., 10.};
+    std::vector<double> actual = evalResult.valuesAsVector();
+    BOOST_CHECK(actual == expected);
+}
+
+BOOST_FIXTURE_TEST_CASE(sum_a_param_as_TS_on_interval_t__t_plus_1__then_square, tests_fixture)
+{
+    // Expression : sum(t .. t+1, p)^2, where p = {p1, p2, p3} = {1., 2., 3.}
+    Node* p = parameter("p", VariabilityType::VARYING_IN_TIME_ONLY);
+
+    Node* from = literal(0);
+    Node* to = literal(1);
+    Node* squared_sum = square(timeSum(from, to, p));
+
+    auto evalResult = evalVisitor->dispatch(squared_sum);
+
+    // Expected evaluation result : ((p1 + p2)^2, (p2 + p3)^2, (p3 + p1)^2)
+    // = (1 + 2)^2, (2 + 3)2, (3 + 1)^2) = (9., 25., 16.)
+    std::vector<double> expected = {9., 25., 16.};
+    std::vector<double> actual = evalResult.valuesAsVector();
+    BOOST_CHECK(actual == expected);
+}
+
+BOOST_FIXTURE_TEST_CASE(sum_a_parameter_as_literal_on_interval_t__t_plus_1, tests_fixture)
+{
+    // Expression : sum(t .. t+1, p), where p = 7
+    Node* seven = literal(7);
+
+    Node* from = literal(0);
+    Node* to = literal(1);
+    Node* sum = timeSum(from, to, seven);
+
+    auto evalResult = evalVisitor->dispatch(sum);
+
+    // Expected evaluation result : p + p = 14
+    BOOST_CHECK_EQUAL(evalResult.valueAsDouble(), 14.);
```

Comment:
Ok in this PR, in all following tests the behavior of the sum operator will change in the future to return a vector ([14, 14, 14] in this case)